kornia/geometry/keypoints.py

```diff
@@ -13,16 +13,34 @@ def _merge_keypoint_list(keypoints: List[Tensor]) -> Tensor:
     raise NotImplementedError
 
 
+def inside_image(coords, image_size):
+    return (
+        (coords[..., 0] >= 0)
+        & (coords[..., 0] < image_size[..., None, 0])
+        & (coords[..., 1] >= 0)
+        & (coords[..., 1] < image_size[..., None, 1])
+    )
+
+
 class Keypoints:
     """2D Keypoints containing Nx2 or BxNx2 points.
 
     Args:
         keypoints: Raw tensor or a list of Tensors with the Nx2 coordinates
         raise_if_not_floating_point: will raise if the Tensor isn't float
+        image_size: (B, W, H) or (W, H) of the image the keypoints are referring to.
```

Comment:
I think having `image_size ` as an init argument is not a good design. Is it easy to remove?

Reply:
Removed. I instead defaulted to the `forward_input_shape` field unless `output_size` was specified.